code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -107,3 +107,54 @@
 	if(shoes && (shoes.item_flags & NOSLIP) && istype(shoes, /obj/item/clothing/shoes/magboots))  //magboots + dense_object = no floating
 		return 1
 	return 0
+
+/mob/living/carbon/human/handle_footstep(atom/T)
+	if(..())
+		if(T.footstep_sounds["human"])
+			var/S = pick(T.footstep_sounds["human"])
+			if(S)
+				if(m_intent == "run")
+					if(!(step_count % 2)) //every other turf makes a sound
+						return 0
+
+				var/range = -(world.view - 2)
+				if(m_intent == "walk")
+					range -= 0.333
+				if(!shoes)
+					range -= 0.333
+
+				//shoes + running
```

Comment:
Don't commit commented code if it's unused or place TODO block if commented functionality is in progress.